spec/rspec/core/suite_hooks_spec.rb

```diff
@@ -23,6 +23,27 @@ module RSpec::Core
           }.to raise_error(ZeroDivisionError)
         end
 
+        it 'runs in the context of an example group' do
+          run_context = nil
+          RSpec.configuration.__send__(registration_method, :suite) { run_context = self }
+          RSpec.configuration.with_suite_hooks { }
+          expect(run_context).to be_a ExampleGroup
+        end
+
+        it 'allows access to rspec-mocks methods within the hook' do
+          run = false
+          RSpec.configuration.__send__(registration_method, :suite) { double('something'); run = true }
```

Comment:
I see why it works now...

* A bare double with no allowances or expectations does not trigger the `The use of doubles or partial doubles from rspec-mocks outside of the per-test lifecycle is not supported. (RSpec::Mocks::OutsideOfExampleError)` error.  So `double('foo')` is OK.
* Even if you change to `double('foo', :msg => "val")` it still passes because there is an active rspec-mocks scope here since this is running inside and rspec example.

For a `before(:suite)` hook declared in `spec_helper`, `double('foo', :msg => "val")` does fail as I would expect.